### PR TITLE
1.0 release notes and authors updates

### PR DESCRIPTION
**Pull request by brson**
cc @aturon

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -1,79 +1,146 @@
...
+  the `PhantomFn` and `MarkerTrait` lang items, which have been
+  removed.
+* ["-" is no longer a valid character in crate names, the `extern crate
+  "foo" as bar` syntax has been replaced with `extern crate foo as
+  bar`, and Cargo now automatically translates "-" in *package* names
+  to "_" for the crate name][cr].
+* [Lifetime shadowing is an error][lt].
+* [`Send` no longer implies `'static`][send-rfc], which made possible
+  the [`thread::scoped` API][scoped]. Scoped threads can borrow data
+  from their parent's stack frame -- safely!
```
Comment (bluss) [inline RELEASES.md]: `thread::scoped` is not in 1.0

File: RELEASES.md [hunk 2]
```diff
@@ -1,79 +1,146 @@
...
+  was the major library focus for this cycle.
+* The path API was [revised][path-normalize] to normalize `.`,
+  adjusting the tradeoffs in favor of the most common usage.
+* A large number of remaining APIs in `std` were also stabilized
+  during this cycle; about 75% of the non-deprecated API surface
+  is now stable.
+* The new [string pattern API][string-pattern] landed, which makes
+  the string slice API much more internally consistent and flexible.
+* A shiny [framework for Debug implementations][debug-builder] landed.
+  This makes it possible to opt in to "pretty-printed" debugging output.
```
Comment (bluss) [inline RELEASES.md]: Debug builders are not stable yet.

File: RELEASES.md [hunk 3]
```diff
@@ -1,79 +1,146 @@
...
+* [`Send` no longer implies `'static`][send-rfc], which made possible
+  the [`thread::scoped` API][scoped]. Scoped threads can borrow data
+  from their parent's stack frame -- safely!
+* [UFCS now supports trait-less associated paths][moar-ufcs] like
+  `MyType::default()`.
+* Primitive types [now have inherent methods][prim-inherent],
+  obviating the need for extension traits like `SliceExt`.
+* Methods with `Self: Sized` in their `where` clause are [considered
+  object-safe][self-sized], allowing many extension traits like
+  `IteratorExt` to be merged into the traits they extended.
```
Comment (bluss) [inline RELEASES.md]: Less important nitpick: These methods are "exempt from object safety checks"?

**Comment by cjwelborn** [comment]
I noticed at least one duplicate. There may be others:

```
GlacJAY <glacjay@gmail.com>
...
Jay True <glacjay@gmail.com>
```

**Comment by nham** [comment]
I'm in the list twice:

```
Nick Hamann <nick@wabbo.org>
nham <hamann.nick@gmail.com>
```

You can remove the second one if you'd like.

**Comment by cjwelborn** [comment]
There are 2 duplicated names (Alfie John and Jakub Wieczorek with different email addresses). I don't know if it's intentional or not, so I haven't tried to make a PR or anything. I found 23 duplicate email addresses (names are different, some use their middle name, some are 'handles').

**Comment by cjwelborn** [comment]
@brson, I made a PR on your repo that removes the duplicates in what I *think* is a sane way to go about it.

**Comment by brson** [comment]
@cjwelborn Thanks! Merged.

**Comment by brson** [comment]
@tshepang Under the normal release schedule, the beta of 4 weeks ago would not have its own release notes - those would be the release notes of the next stable release. So the thinking was to combine them to provide the full picture of what happened recently. The amount of new work that went into 1.0 this cycle was actually pretty small...

Beta's are not 'releases'.

**Comment by alfiedotwtf** [comment]
Looks like I'm in there twice. Thanks @cjwelborn, but my preferred email for all things Rust is alfie@alfie.wtf :)

**Comment by cjwelborn** [comment]
@alfiedotwtf, sorry about that. I'm glad you caught it before it was finalized.

**Comment by daramos** [comment]
D'oh! Looks like I had my email as my name in gitconfig. 

Any chance `dan@daramos.com <dan@daramos.com>` could be changed to `Daniel Ramos <dan@daramos.com>`?

Sorry about that...

**Comment by caipre** [comment]
If you're taking these last minute comments... `s/caipre/Nick Platt`

File: RELEASES.md [hunk 4]
```diff
@@ -1,79 +1,146 @@
...
+* A [Reflect trait][reflect] was introduced, which means that
+  downcasting via the `Any` trait is effectively limited to concrete
+  types. This helps retain the potentially-important "parametricity"
+  property: generic code cannot behave differently for different type
+  arguments except in minor ways.
+* The `unsafe_destructor` feature is now deprecated in favor of the
+  [new `dropck`][dropck]. This change is a major reduction in unsafe
+  code.
+* Trait coherence was [revised again][fundamental], this time with an
+  eye toward API evolution over time.
```
Comment (sanxiyn) [inline RELEASES.md]: This seems to be a duplicate.

**Comment by rpjohnst** [comment]
Could you update "Russell <rpjohnst@gmail.com>" to "Russell Johnston <rpjohnst@gmail.com>"?

**Comment by Zoxc** [comment]
Can someone move John Kleint out of my names?

File: RELEASES.md [hunk 5]
```diff
@@ -1,79 +1,146 @@
...
+
+Language
+--------
+
+* Several [restrictions have been added to trait coherence][coh] in
+  order to make it easier for upstream authors to change traits
+  without breaking downsteam code.
+* Digits of binary and octal literals are [lexed more eagerly][lex] to
+  improve error messages and macro behavior. For example, `0b1234` is
+  now lexed as `0b1234` instead of two tokens, `0b1` and `34`.
```
Comment (sfackler) [inline RELEASES.md]: The `2` appears to be missing.

**Comment by ogham** [comment]
Not sure why I decided to abbreviate my name. Could you change 'Ben S' to 'Ben Sago'? Ta

**Comment by thorncp** [comment]
Please change `Chris Thorn <thorn@thoughtbot.com>` to `Chris Thorn <chris@thorn.co>`.

So much thanks to the Rust team for doing this! :heartpulse:

**Comment by sfackler** [comment]
@ogham "Ben S" is the name associated with your github profile: https://github.com/ogham

**Comment by laumann** [comment]
@brson Is it any kind of code contribution that counts (ie commits in master)?

**Comment by TyOverby** [comment]
Could you change "TyOverby" to "Ty Overby"?  Thanks!

**Comment by huonw** [comment]
@laumann yes!

**Comment by blackbeam** [comment]
Could you please change "blackbeam" to "Anatoly Ikorsky"?

**Comment by laumann** [comment]
@huonw Awesome, then I'm missing from the list:
`Thomas Bracht Laumann Jespersen <laumann.thomas@gmail.com>`. 

(see d14109ec7e90f42a7cb966415b96094b146d3706)

**Comment by huonw** [comment]
@laumann, oh, I apologise because I realise I may have misspoke: this is the first release where the release *isn't* from the master branch, so some recent contributions may not be in it (I'm unsure if that commit is).

**Comment by laumann** [comment]
@huonw Oh, ok, that might be why - is it from the beta branch?

**Comment by eddyb** [comment]
Minor thing: my full name is `Eduard-Mihai Burtescu` and the hyphen makes it so I can't omit `Mihai` in legal contexts.
I'm nor it matters here (for copyright/licensing?), but I thought I'd mention it.

**Comment by MrFloya** [comment]
After my email got spammed after that blogpost I created a gh alias.
It would be great if you could change `Florian Wilkens floya@live.de` to `Florian Wilkens mrfloya_github@outlook.com`. Thanks a lot!

File: AUTHORS.txt [hunk 6]
```diff
@@ -970,12 +977,14 @@ nwin <nwin@users.noreply.github.com>
 oli-obk <github6541940@oli-obk.de>
```
Comment (oli-obk) [inline AUTHORS.txt]: remove this please and change the email of `Oliver Schneider` to this email address

**Comment by oli-obk** [comment]
> my email got spammed after that blogpost

I've also noticed this. I'm not sure how to handle this, as these are added automatically and the owner of the email address doesn't get a heads-up that this will happen. I don't want to complain or critizise, but I'd like to discuss how to handle this in the future. Automated email farmers don't seem to detect github commit authors, but they read through blog posts and files uploaded to github.

**Comment by daboross** [comment]
If it wouldn't be any trouble, changing `Dabo Ross` to `David Ross` would be nice. Email is correct.

File: RELEASES.md [hunk 7]
```diff
@@ -1,79 +1,146 @@
...
+Libraries
+---------
+
+* The `thread_local` module [has been renamed to `std::thread`][th].
+* The methods of `IteratorExt` [have been moved to the `Iterator`
+  trait itself][ie].
+* Several traits that implement Rust's conventions for type
+  conversions, `AsMut`, `AsRef`, `From`, and `Into` have been
+  [centralized in the `std::convert` module][con].
+* The `FromError` trait [was removed in favor of `From`][fe].
```
Comment (killercup) [inline RELEASES.md]: Link doesn't render, below you define `[ef]` instead of `[fe]`.

**Comment by nwin** [comment]
Please replace nwin with `Nils Winter <nils.winter@gmail.com>`

**Comment by alfiedotwtf** [comment]
@oli-obk hiding email addresses in git logs as opposed to blog posts is security through obscurity. It's either secret or not secret, there is no middle ground.

**Comment by pelmers** [comment]
I'd like to change my email from `peter.elmers@yahoo.com` to `peter.elmers@rice.edu`. Thanks!

**Comment by m-r-r** [comment]
Hello,
Could you please change `m-r-r <raybaudroigm@gmail.com>` to `Mickaël Raybaud-Roig <raybaudroigm@gmail.com>` ?

**Comment by inrustwetrust** [comment]
Hi, could you please change inrustwetrust to John Talling?

**Comment by iancormac84** [comment]
Hello. Could you change iancormac84 to Nathan Wilson please?

**Comment by hjr3** [comment]
Hello,
Could you please change `Herman J. Radtke III <hermanradtke@gmail.com>` to `Herman J. Radtke III <herman@hermanradtke.com>` ?

**Comment by bguiz** [comment]
@huonw 

> this is the first release where the release isn't from the master branch, so some recent contributions may not be in it (I'm unsure if that commit is).

If my minor contributions count, I'd like to be added:

- https://github.com/rust-lang/rust/pull/24894
- https://github.com/rust-lang/rust/pull/25054
- https://github.com/rust-lang/rust/pull/25055
- https://github.com/rust-lang/rust/pull/25068

**Comment by quantheory** [comment]
@bguiz This is for the release which was branched last month, so your changes probably won't be credited until 1.1. (In six weeks? I can't remember whether 1.1 is planned to take 1 or 2 cycles.) Though maybe the doc changes *should* be cherry-picked for 1.0, if they weren't already.

That said, I'm sure that it does count, since I'm credited and my contributions were probably even more minor before the 1.0 beta was branched.

**Comment by graydon** [comment]
You should probably change my email address to graydon@pobox.com, pretty sure the Mozilla one bounces now.

(Braces for uptick in spam)

**Comment by bguiz** [comment]
@quantheory Thanks for the info. Yeah, I definitely think that all of those patches should be back-merged to 1.0, as they are definitely applicable. I'll leave it up to the reviewers/ core team to decide though.

**Comment by toffaletti** [comment]
@brson Please update my email to toffaletti@gmail.com, jason@topsy.com no longer exists.

**Comment by jakub-** [comment]
By the way, `.mailmap` is where the correct e-mail mappings should go so that `git log` lists the right one out of the box.

**Comment by steveklabnik** [comment]
@bguiz generally speaking, i'm going to backport the entire book right before release. You should get credit in this file, for sure. /cc @brson

**Comment by steveklabnik** [comment]
@brson: I'm going through old PRs, here's a list of people who should be in for doc stuff. I didn't check to see if they were here or not, but they should be:

- @poga for https://github.com/rust-lang/rust/pull/24982
- @carols10cents for https://github.com/rust-lang/rust/pull/24981
- @hech for https://github.com/rust-lang/rust/pull/24970
- @nham for https://github.com/rust-lang/rust/pull/24961
- @tynopex for https://github.com/rust-lang/rust/pull/24960
- @oli-obk for https://github.com/rust-lang/rust/pull/24933
- @tcard for https://github.com/rust-lang/rust/pull/24891
- @mbrubeck for https://github.com/rust-lang/rust/pull/24871
- @dhardy for https://github.com/rust-lang/rust/pull/24863
- @barometz for https://github.com/rust-lang/rust/pull/24862
- @diwic for https://github.com/rust-lang/rust/pull/24844
- @gbgamer for https://github.com/rust-lang/rust/pull/24842
- @heejongahn for https://github.com/rust-lang/rust/pull/24815
-  @gareins for https://github.com/rust-lang/rust/pull/24849
- @aramvisser for https://github.com/rust-lang/rust/pull/24496
- @jooert for https://github.com/rust-lang/rust/pull/24783

If I missed anyone else, sorry! Trying to do best effort to pick up people who may have fallen through the cracks here.

**Comment by michal-czardybon** [comment]
This is me:
Michał Czardybon <mczard@poczta.onet.pl>

**Comment by barometz** [comment]
Initially figured I wouldn't bother since I just have the single one-line doc commit, but since I was mentioned, I would like to state for the literal record:

`Dominic van Berkel <dominic@baudvine.net>`

**Comment by steveklabnik** [comment]
> since I just have the single one-line doc commit

All patches are valuable, no matter how small :)

**Comment by poga** [comment]
Thanks for mentioning me as I just fixed a single "\n" in the doc :)

This is me: 
```Poga Po <poga.bahamut@gmail.com>```

**Comment by johannhof** [comment]
Made #25286

`Johann Hofmann <mail@johann-hofmann.com>`

Thanks!

**Comment by DrKwint** [comment]
Steve told me to post here about #25288 

Paul Quint <DrKwint@gmail.com>

Thank you!

**Comment by jxs** [comment]
thanks for the credit folks, can the email on https://github.com/brson/rust/blob/ea0c1ed71ef359b67ff56b19a667e8feebde5dd7/AUTHORS.txt#L965 just be changed to hello@jxs.pt?

thanks!

File: RELEASES.md (hunk 1 above)
Comment (brson) [inline RELEASES.md]: Thanks, fixed.

File: RELEASES.md (hunk 5 above)
Comment (brson) [inline RELEASES.md]: Good eye.

**Comment by brson** [comment]
@Zoxc I didn't understand your comment.

**Comment by brson** [comment]
@laumann Anybody with commits to this repo.

File: RELEASES.md (hunk 7 above)
Comment (brson) [inline RELEASES.md]: Nice catch.

**Comment by brson** [comment]
Updated with all comments addressed (I think) except @Zoxc's.

File: RELEASES.md [hunk 8]
```diff
@@ -1,79 +1,140 @@
...
+  now lexed as `0b1234` instead of two tokens, `0b1` and `234`.
+* Trait bounds [are always invariant][inv], eleminating the need for
+  the `PhantomFn` and `MarkerTrait` lang items, which have been
+  removed.
+* ["-" is no longer a valid character in crate names][cr], the `extern crate
+  "foo" as bar` syntax has been replaced with `extern crate foo as
+  bar`, and Cargo now automatically translates "-" in *package* names
+  to underscore for the crate name.
+* [Lifetime shadowing is an error][lt].
+* [`Send` no longer implies `'static`][send-rfc], which made possible
```
Comment (bluss) [inline RELEASES.md]: `, which made possible` is a half sentence hanging here.

Comment (brson) [inline RELEASES.md]: Fixed.

**Comment by nathanielherman** [comment]
Could my email be changed from nherman @ college.harvard.edu to nherman @ post.harvard.edu?

**Comment by Zoxc** [comment]
@brson My complaint was addressed atleast. I'm no longer listed twice.

File: RELEASES.md [hunk 9]
```diff
@@ -1,79 +1,140 @@
-Version 1.0.0-beta (April 2015)
--------------------------------------
-
-* ~1100 changes, numerous bugfixes
```
Comment (tshepang) [inline RELEASES.md]: am curious why remove change count

Comment (heejongahn) [inline RELEASES.md]: Thanks for mentioning me!

This is me:
Heejong Ahn <heejongahn@gmail.com <poga.bahamut@gmail.com>>

p.s) To be frank, it's my first (and misc) open-source contribution and I
don't know what's really going on here so I just mocked what Poga's mail..
Plz let me know if I've done anything wrong.

2015-05-11 14:28 GMT+09:00 Tshepang Lekhonkhobe <notifications@github.com>:

**Comment by gareins** [comment]
I am:
Ožbolt Menegatti
<ozbolt.menegatti@gmail.com>

If it needs to be ascii, just use 'z' instead of 'ž'. 

ps. Thanks everyone for an awesome language :)

**Comment by jest** [comment]
If spellchecking is considered relevant contribution, please add

```
Przemek Wesołek <jest@go.art.pl>
````

due to https://github.com/rust-lang/rust/pull/24812

**Comment by KokaKiwi** [comment]
If writing a postponed RFC is considered as a relevant contribution (maybe not, but I wanted to ask anyway), is it possible to add
```
KokaKiwi <kokakiwi@kokakiwi.net>
```
Due to https://github.com/rust-lang/rfcs/pull/257

**Comment by alexcrichton** [comment]
r=me whenever this is ready @brson

**Comment by brson** [comment]
I've already picked this into beta, so setting tag to beta-accepted

File: RELEASES.md (hunk 9 above)
Comment (tshepang) [inline RELEASES.md]: @brson I see you put the count back in 94caeb326c45bd0a108abe92ae50f9deda0aeb4c... kool.

**Comment by dhardy** [comment]
I checked the authors file. It contains an old spam-infested email address, which is fine. Now somehow my github email appears in the 1.0 announcement, and right on queue I get a bunch of phishing emails. How'd that happen? I'm now going to have to replace the github email... don't put the new one in blog posts please.

**Comment by alfiedotwtf** [comment]
@dhardy that's security through obscurity. All it takes is a bot to clone one of your repo to see your email address. Consider using temporary aliases for anything public facing.

**Comment by dhardy** [comment]
Consider using temporary aliases for anything public facing.

That's precisely what it is, and mostly my system works (it seems bots are not in the habit of cloning repos). I only get spam on the gmail address and if anyone emails me there, they get an "out-of-office" reply notifying them that I don't read it frequently.

**Pull request merged**